### PR TITLE
Tightened train timing and added per-batch sample rate

### PR DESCRIPTION
**Pull request by RedbackThomson**
Moved `train()` timing to just before and after epoch - no longer times the save_checkpoint.

Added sample rate average meter per-batch for debugging of ongoing performance differences.

### CODE REVIEW

**Comment by RedbackThomson** [comment]
Increased epoch averages from <540 to ~600 on an FP32 with batch size 1024 and 10000 samples.

**Comment by RedbackThomson** [comment]
Increased epoch averages to ~2000 by ignoring initial data loading first batch.

File: pytorch_examples/imagenet/main.py [hunk 1]
```diff
@@ -252,15 +257,25 @@ def train(train_loader, model, criterion, optimizer, epoch):
...
         end = time.time()
 
+        ### Measure sample speed
+        sample_speed.update(sample_count/batch_time.val)
+
         if i % args.print_freq == 0:
+            ### Added sample rate print
             print('Epoch: [{0}][{1}/{2}]\t'
                   'Time {batch_time.val:.3f} ({batch_time.avg:.3f})\t'
+                  'Speed: {sample_speed.val:.3f} ({sample_speed.avg:.3f}) \t'
```
Comment (surajkota) [inline pytorch_examples/imagenet/main.py]: batch speed

resnet50_synthetic_p3_16xlg_runner -> metric policy

**Comment by RedbackThomson** [comment]
Created new metric.

Invocation script will require consuming this new metric - no longer shared with other (chainer and mxnet) synthetic runs.

**Comment by RedbackThomson** [comment]
Reverted epoch timing back to include the first batch, but now batches will only print the `BatchSpeed` metric for batch 2+.

File: task_config_template.cfg [hunk 2]
```diff
@@ -341,6 +341,11 @@ patterns = ['images/sec: (\d+\.\d+)']
 metrics = ['Speed(Images/sec)']
 compute_method = ['average']
 
+[pytorch_resnet50_synthetic_p3_16xlg_runner]
+patterns = ['BatchSpeed (\d+\.\d+|\d+)', 'Time cost=(\d+\.\d+|\d+)']
```
Comment (surajkota) [inline task_config_template.cfg]: this should have both BatchSpeed and EpochSpeed

```
[resnet50_synthetic_p3_16xlg_runner]
patterns = ['Speed: (\d+\.\d+|\d+) samples/sec', 'BatchSpeed (\d+\.\d+|\d+)', 'Time cost=(\d+\.\d+|\d+)']
metrics = ['speed(samples/sec)', 'batchspeed(samples/sec)','Time']
compute_method = ['average', 'average', 'last']
```

let me know if it makes sense

File: pytorch_examples/imagenet/main.py [hunk 3]
```diff
@@ -52,8 +52,9 @@
                     help='momentum')
 parser.add_argument('--weight-decay', '--wd', default=1e-4, type=float,
                     metavar='W', help='weight decay (default: 1e-4)')
-parser.add_argument('--print-freq', '-p', default=10, type=int,
-                    metavar='N', help='print frequency (default: 10)')
+### Modified for more verbose outputs by default
+parser.add_argument('--print-freq', '-p', default=1, type=int,
```
Comment (surajkota) [inline pytorch_examples/imagenet/main.py]: lets change it in the lambda rather than here

**Comment by RedbackThomson** [comment]
With most recent changes invocation script will require changing the run arguments to `-p 1` for best results.

**Review by surajkota — APPROVED** [review]
